x-pack/platform/packages/shared/ai-assistant/ai-assistant-connector-selector-action/connector_selector.tsx

```diff
@@ -0,0 +1,190 @@
+  const [uncontrolledValue, setUncontrolledValue] = useState<string | undefined>(() => {
+    if (props.defaultValue !== undefined) return props.defaultValue;
+    return undefined;
+  });
+
+  const selectedValue = isControlled ? controlledValue : uncontrolledValue;
+
+  const defaultConnectorBadge = useMemo(() => {
+    return (
+      <EuiBadge color="hollow" data-test-subj="defaultConnectorBadge">
+        {i8n.defaultConnectorLabel}
+      </EuiBadge>
+    );
+  }, []);
+
+  const preConfiguredConnectors = useMemo(
+    () =>
+      props.preConfiguredConnectors.map((connector) => ({
+        ...connector,
+        key: connector.value,
+        checked: connector.value === selectedValue ? 'on' : undefined,
+        append: connector.value === props.defaultConnectorId ? defaultConnectorBadge : undefined,
+      })),
+    [props.preConfiguredConnectors, selectedValue, props.defaultConnectorId, defaultConnectorBadge]
+  );
+
+  const customConnectors = useMemo(
+    () =>
+      props.customConnectors.map((connector) => ({
+        ...connector,
+        key: connector.value,
+        checked: connector.value === selectedValue ? 'on' : undefined,
+        append: connector.value === props.defaultConnectorId ? defaultConnectorBadge : undefined,
+      })),
+    [props.customConnectors, selectedValue, props.defaultConnectorId, defaultConnectorBadge]
+  );
+
+  const options = useMemo<EuiSelectableOption<T>[]>(
+    () => [
+      ...(preConfiguredConnectors.length > 0
+        ? [
+            {
+              label: i8n.preConfiguredConnectorLabel,
+              isGroupLabel: true,
+              value: 'preConfiguredLabel',
+            } as EuiSelectableOption<T>,
+          ]
+        : []),
+      ...preConfiguredConnectors,
+      ...(customConnectors.length > 0
+        ? ([
+            { label: i8n.customConnectorLabel, isGroupLabel: true, value: 'customLabel' },
+          ] as EuiSelectableOption<T>[])
+        : []),
+      ...customConnectors,
+    ],
+    [preConfiguredConnectors, customConnectors]
+  );
+
+  const handleChange = (newOptions: EuiSelectableOption<T>[]) => {
+    const selectedOption = newOptions.find((option) => option.checked === 'on');
+    if (!selectedOption) return;
+
+    if (!isControlled) {
+      setUncontrolledValue(String(selectedOption.value));
+    }
+
+    props.onValueChange?.(String(selectedOption.value), selectedOption);
+  };
+
+  return (
+    <EuiPanel
+      hasShadow={false}
+      borderRadius="none"
+      paddingSize="none"
+      grow={false}
+      data-test-subj={props['data-test-subj'] ?? 'aiAssistantConnectorSelector'}
+    >
+      <EuiSelectable
+        aria-label={i8n.selectableAriaLabel}
+        singleSelection
+        options={options}
+        onChange={handleChange}
+      >
+        {(list) => (
+          <>
+            {list}
+            {(props.onAddConnectorClick || props.onManageConnectorsClick) && (
+              <EuiPopoverFooter paddingSize="s">
```

Comment:
nit: I would split this out into its own component.  The two props determine if it's rendered, so they would be a part of that footer component, with this selector extending them.  It would encapsulate the concerns, make it easier to read/grok:

```tsx
<EuiPanel ...>
  <EuiSelectable ...>
    {(list) => (
      <>
        {list}
        <ConnectorSelectorFooter {...rest} /> // might not render
      </>
    )}
  </EuiSelectable>
</EuiPanel>
```

You could also consider exposing a `footer` prop, or `children`, and allow consumers to provide a footer, then wrap it all up in a default... it depends on where and how it will be consumed.